frontend/web/src/app/stl-viewer/classes/Scene.ts

```diff
@@ -0,0 +1,114 @@
+import { AABB } from "./AABB";
+
+/**
+	@class Scene
+
+	This class implements scene that contains a group of meshes that forms the world.
+ */
+export class Scene {
+    /**
+     * {String} Name of the scene.
+     */
+    name: string = '';
+    /**
+     * {String} Source URL of the scene, empty if none. Read only.
+     */
+    srcUrl:string = '';
+    /**
+     * {JSC3D.AABB} The Axis-aligned bounding box of the whole scene. Read only.
+     */
+    aabb: AABB = null;
+    children = null;
+    maxChildId: number = 1;
+
+    constructor(name) {
+        this.name = name || '';
+        this.srcUrl = '';
+        this.aabb = null;
+        this.children = [];
+        this.maxChildId = 1;
+    };
+
+    /**
+        Initialize the scene.
+    */
+    init() {
+        if(this.isEmpty())
+            return;
+
+        for(var i=0; i<this.children.length; i++)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') for statements must be braced [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS121)
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Forbidden 'var' keyword, use 'let' or 'const' instead [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS3504)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Expected a 'for-of' loop instead of a 'for' loop with this simple iteration [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS4138)